components/director/internal/domain/application/service.go

```diff
@@ -267,3 +285,16 @@ func (s *service) deleteRelatedResources(applicationID string) error {
 
 	return nil
 }
+
+func validateApplicationName(name string) (bool, error) {
+	match, err := regexp.MatchString("^[a-z\\.-]+$", name)
```

Comment:
your regexp does not match what is described here: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
for example max length of name. 
Please check if you can use validation from k8s itself: https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/apps/validation/validation.go#L379